src/cljs/rems/util.cljs

```diff
@@ -211,4 +208,19 @@
       nil -1
       nil nil
       nil {}
-      nil [])))
\ No newline at end of file
+      nil [])))
+
+(defn escape-element-id [id]
+  (when (string? id)
+    (-> id
+        (str/replace #"\s" "")
+        (str/replace #"[^A-Za-z0-9\-_]" "_")
+        (str/replace-first #"^[0-9]" #(str "id_" %)))))
+
+(deftest test-escape-element-id
+  (testing "should replace element special characters with underscore"
+    (is (nil? (escape-element-id nil)))
+    (is (= "elementid-123" (escape-element-id "element id-123 ")))
+    (is (= "element_id_123" (escape-element-id "element.id#123")))
+    (is (= "id_123-element" (escape-element-id " 123-element")))))
```

Comment:
I don't quite understand the escaping logic here. Should we rather escape ` ` with `_` and not trim?

Reply:
`-` is a valid id character, stated by at least MDN: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id
> Technically, in [HTML5](https://developer.mozilla.org/en-US/docs/Glossary/HTML5), the value for an id attribute may contain any character, except [whitespace](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace) characters. However, to avoid inadvertent errors, only [ASCII](https://developer.mozilla.org/en-US/docs/Glossary/ASCII) letters, digits, '_', and '-' should be used and the value for an id attribute should start with a letter.

Reply:
refactored the function so it replaces whitespace with underscore instead of trim

Reply:
looking at `rems.test-util/test-linkify` the original usage seems to have been more akin to string than hiccup:
```clojure
(testing "retain original string"
      (is (= (apply str (linkify "a b c") "a b c"))))
```